source/puppet/5.0/lang_data_hash.md

```diff
@@ -29,17 +29,25 @@ notice($result)
 
 ## Syntax
 
-Hashes are written as a pair of curly braces containing any number of key/value pairs. A key is separated from its value by a `=>` (arrow, fat comma, or hash rocket), and adjacent pairs are separated by commas. An optional trailing comma is allowed between the final value and the closing curly brace.
+Hashes are written as a pair of curly braces containing any number of key/value pairs. Keys should be quoted with single or double quotes. A key is separated from its value by a `=>` (arrow, fat comma, or hash rocket), and adjacent pairs are separated by commas. An optional trailing comma is allowed between the final value and the closing curly brace.
```

Comment:
Since keys of hashes can be anything, saying that they should be quoted is not correct.

Reply:
It is however correct to say that "It is best if keys *that are strings* are quoted"